lib/session/call-session.js

```diff
@@ -2232,17 +2245,59 @@ Duration=${duration} `
   }
 
   /**
-   * Handle incoming REFER if we are in a dial task
+   * Handle incoming REFER
    * @param {*} req
    * @param {*} res
    */
   _onRefer(req, res) {
     const task = this.currentTask;
     const sd = task.sd;
-    if (task && TaskName.Dial === task.name && sd) {
+    if (task && TaskName.Dial === task.name && sd && task.referHook) {
       task.handleRefer(this, req, res);
     }
     else {
+      this._handleRefer(req, res);
```

Comment:
this almost entirely duplicates code that is in the Dial task.  Can it be refactored into a common utils file?

Reply:
They are not identical. the handleRefer from dial verb is handling for both legs which is totally different from call-session handle for incoming call leg only.